### PR TITLE
fix demo-bug

### PR DESCRIPTION
**Pull request by anntisa**
when using demo mode, the surveywall would not open

### CODE REVIEW

File: Survata/Survey.swift [hunk 1]
```diff
@@ -364,7 +364,7 @@ class SurveyViewController: UIViewController {
 					self?.dismissViewControllerAnimated(true, completion: nil)
 					self?.onCompletion?(.CreditEarned)
 				}
-			} else {
+			} else if (!(self?.survey?.option.testing ?? false) && (self?.survey?.option.preview ?? "").isEmpty) {
```
Comment (anntisa) [inline Survata/Survey.swift]: if 'testing' is nil or false return true
if 'preview' is nil or "" return true

both must be true to get rid of the surveywall

Comment (anntisa) [inline Survata/Survey.swift]: and return 'NoSurveyAvailable'

Comment (jameswwarner) [inline Survata/Survey.swift]: What does this mean: self?.survey?.option.testing ?? false

Comment (anntisa) [inline Survata/Survey.swift]: if testing is empty/nil, then it will set the value as false,

Comment (rogueleaderr) [inline Survata/Survey.swift]: Actually as I'm looking at the surrounding logic...what is the situation when we'd get to this code and have `data` be nil when we're also not previewing or testing?

Comment (anntisa) [inline Survata/Survey.swift]: as far as iOS goes, i've not seen #2 happen. i've only seen it in Android

File: Survata/Survey.swift [hunk 2]
```diff
@@ -356,6 +356,9 @@ class SurveyViewController: UIViewController {
 
 		surveyView.on("load") {[weak self] data in
 			self?.survey.print("data \(data)")
+			let testing = self?.survey?.option.testing ?? false
+			let previewOption = (self?.survey?.option.preview ?? "")
+			let preview = previewOption.isEmpty
```
Comment (rogueleaderr) [inline Survata/Survey.swift]: sorry I think my last comment had a flipped ! sign. 

Are we in preview mode when the previewOption is empty, or when it has a string? (I would assume that we *are* previewing if the string is not empty, so then the line should be `let preview = !previewOption.isEmpty`)

And then do we want to dismiss the survey when we are previewing or when we aren't previewing?

Comment (anntisa) [inline Survata/Survey.swift]: we want to dismiss the survey is the preview is empty and testing is false

Comment (anntisa) [inline Survata/Survey.swift]: if any of those 2 variables have a 'truthy' value in it, for example
``` 
preview = '5fd725139884422e9f1bb28f776c702d'
testing = true
```
then it's a demo survey and we dont want to dismiss the surveywall view

Comment (rogueleaderr) [inline Survata/Survey.swift]: okay so then the lines should be 

`let preview = !previewOptions.isEmpty`

then 

`if !testing && !preview`

**Comment by rogueleaderr** [comment]
:ship: 🎆

**Pull request merged**